### PR TITLE
benchmarks/dynamo: fix timm model load for models without drop_rate support

### LINKED ISSUE #187301: benchmarks/dynamo/timm_models.py: model loading fails for models that don't accept drop_rate
### Problem description:
Running the timm benchmark with certain newer models (e.g. timm/gemma4_vit_570m_enc.gemma4_31b_it) fails at model load time with:
```
Gemma4VitEncoder.__init__() got an unexpected keyword argument 'drop_rate'
```

### Reproduction
```
python \
    benchmarks/dynamo/timm_models.py --performance --inference --bfloat16 \
    --backend=inductor --output=timm_models_inference.csv --device xpu \
    --only timm/gemma4_vit_570m_enc.gemma4_31b_it
```
Or in isolation:
```
from timm.models import create_model
create_model(
    'timm/gemma4_vit_570m_enc.gemma4_31b_it', in_chans=3, scriptable=False,
    num_classes=None, drop_rate=0.0, drop_path_rate=None, drop_block_rate=None,
    pretrained=True,
)
```

### Root Cause
`_download_model` in `benchmarks/dynamo/timm_models.py` hardcodes `drop_rate`, `drop_path_rate`,
and `drop_block_rate` kwargs for all models. Newer timm model architectures (like Gemma4VitEncoder) do not accept these kwargs.

### Possible fix:
Retry `create_model` without the drop kwargs on TypeError:
```
@download_retry_decorator
def _download_model(self, model_name):
    try:
        model = create_model(
            model_name,
            in_chans=3, scriptable=False, num_classes=None,
            drop_rate=0.0, drop_path_rate=None, drop_block_rate=None,
            pretrained=True,
        )
    except TypeError:
        model = create_model(
            model_name,
            in_chans=3, scriptable=False, num_classes=None,
            pretrained=True,
        )
    return model
```

cc @mruberry @chauhang @penguinwu

### PR DESCRIPTION
**Pull request by ashahba**
### Summary
Fixes #187301

Some newer timm models (e.g. `Gemma4VitEncoder` from `timm/gemma4_vit_570m_enc.gemma4_31b_it`) do not accept `drop_rate`, `drop_path_rate`, or `drop_block_rate` kwargs in their `__init__`.
The benchmark hardcoded these for all models, causing a `TypeError` at load time:
```
Gemma4VitEncoder.__init__() got an unexpected keyword argument 'drop_rate'
```
The fix retries `create_model` without the drop kwargs on `TypeError`, preserving existing behavior for models that do accept them.

### Test plan
- Reproduced the error on tip of main before the fix
- Verified the model loads successfully after the fix

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo @azahed98

### CODE REVIEW

**Comment by jansel** [comment]
@claude review these changes

**Comment by ashahba** [comment]
> **Claude finished @jansel's task in 48s** —— [View job](https://github.com/pytorch/pytorch/actions/runs/27555579490)
> 
> ### Code Review
> I reviewed the change in `benchmarks/dynamo/timm_models.py`. The fix is reasonable and well-targeted at the reported bug. A few points for consideration:
> 
> **Tasks**
> 
> * [x]  Read PR diff and surrounding code
> * [x]  Analyze the fix
> * [x]  Post review feedback
> 
> #### Overall
> The approach (retry `create_model` without the drop kwargs on `TypeError`) is a pragmatic fix for models like `Gemma4VitEncoder` that don't accept `drop_rate`/`drop_path_rate`/`drop_block_rate`. Since this is benchmark-only code, the bar is lower than for core library code, and the change preserves existing behavior for models that do accept these kwargs.
> 
> #### Suggestions
> 1. **`except TypeError` is broad** (`benchmarks/dynamo/timm_models.py:227`). A `TypeError` raised for _any_ reason inside `create_model` (e.g. a genuine bug in model construction, or pretrained-weight loading) will be silently swallowed and retried without the drop kwargs. That can mask a real error and/or produce a silently-misconfigured model. Consider narrowing to the actual signature mismatch, e.g.:
>    ```python
>    except TypeError as e:
>        if "unexpected keyword argument" not in str(e):
>            raise
>        model = create_model(...)
>    ```
> 2. **Silent behavior change with no signal.** When the fallback fires, the model is built without the drop kwargs and nothing is logged. A `log.warning(...)` (the module already has a `logging` import / logger pattern in the benchmark harness) would make it clear in benchmark output that a model ran with different regularization settings than the rest of the suite.
> 3. **Duplicated `create_model` call.** The two calls differ only by the three drop kwargs. Building a kwargs dict avoids the duplication and keeps the common args in one place:
>    ```python
>    kwargs = dict(
>        in_chans=3,
>        scriptable=False,
>        num_classes=None,
>        pretrained=True,
>    )
>    try:
>        model = create_model(model_name, drop_rate=0.0, drop_path_rate=None, drop_block_rate=None, **kwargs)
>    except TypeError as e:
>        if "unexpected keyword argument" not in str(e):
>            raise
>        model = create_model(model_name, **kwargs)
>    ```
> 
> None of these are blocking; (1) is the one I'd most encourage, since a blanket `except TypeError` in benchmark loading code can hide unrelated failures. The CI failures on this PR (`mobilenetv2_100`, `shufflenet_v2_x1_0`, `tf_efficientnet_b0`) are flagged as flaky/broken-trunk and unrelated to this change.
> 
> [Fix this →](https://claude.ai/code?q=In%20benchmarks%2Fdynamo%2Ftimm_models.py%20_download_model%20on%20branch%20fix%2Ftimm-drop-rate-187301%2C%20narrow%20the%20except%20TypeError%20to%20only%20retry%20when%20the%20message%20contains%20%22unexpected%20keyword%20argument%22%20(re-raise%20otherwise)%2C%20add%20a%20log.warning%20when%20the%20fallback%20fires%2C%20and%20deduplicate%20the%20two%20create_model%20calls%20using%20a%20shared%20kwargs%20dict.&repo=pytorch/pytorch) · branch `fix/timm-drop-rate-187301`

@claude feedback is reasonable and all have been addressed in the recent commit.

**Review by malfet — APPROVED** [review]

**Comment by malfet** [comment]
@ashahba and can you sign the CLA?

**Review by jansel — APPROVED** [review]

**Comment by ashahba** [comment]
> @ashahba and can you sign the CLA?

Done!
Feel free to move forward with merging!
Thanks.

**Comment by jansel** [comment]
@claude explain those CI failures, are they related to this PR?

**Comment by ashahba** [comment]
@jansel Is there any actions that need to be taken on my side to move this PR to merge ready state?

**Comment by jansel** [comment]
@claude explain those CI failures, are they related to this PR?

**Comment by jansel** [comment]
@claude explain those CI failures, are they related to this PR?

**Comment by jansel** [comment]
@claude explain those CI failures, are they related to this PR?
